docusaurus/docs/tutorials/converting-a-frontend-datasource-to-backend.md

```diff
@@ -0,0 +1,370 @@
+	}
+
+	// Create frame and add it to the response
+	dataResp := backend.DataResponse{
+		Frames: []*data.Frame{
+			data.NewFrame(
+				"response",
+				data.NewField("time", nil, times),
+				data.NewField("values", nil, values),
+			),
+		},
+	}
+	return dataResp, err
+}
+```
+
+### Other resource requests
+
+Finally, there is a third type of request that plugins may implement even though it's optional. This is what we call "resources". Resources are additional endpoints that the plugin can expose and use to populate the query or config editors. For example, a resource can be used to populate a dropdown with a list of available tables in a database.
+
+In a frontend data source, the plugin should define the resources in the `plugin.json` file as `routes` and use the `fetch` method to get the data. For example (for simplicity, authentication is omitted in this example):
+
+```json title="src/plugin.json"
+{
+  "routes": [
+    {
+      "path": "tables",
+      "url": "https://api.example.com/api/v1/tables",
+      "method": "GET"
+    }
+  ]
+}
+```
+
+```typescript title="src/DataSource.ts"
+export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
+  async getTables() {
+    const response = await lastValueFrom(
+      getBackendSrv().fetch<MetricsResponse>({
+        url: `${this.url}/tables`,
+        method: 'GET',
+      })
+    );
+    return response.data;
+  }
+}
+```
+
+For a backend data source, the plugin should implement the `CallResourceHandler` interface. This interface should handle the different possible resources. For example:
+
+```go title="pkg/plugin/datasource.go"
+func NewDatasource(_ context.Context, _ backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	return &Datasource{
+		CallResourceHandler: newResourceHandler(),
+	}, nil
+}
+
+func newResourceHandler() backend.CallResourceHandler {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/tables", handleTables)
+
+	return httpadapter.New(mux)
+}
+
+func handleTables(w http.ResponseWriter, r *http.Request) {
+  // Get tables
+  res, err :=	http.DefaultClient.Get("https://api.example.com/api/v1/tables")
+  // Handle errors (omited)
+	body, err := io.ReadAll(res.Body)
+  // Handle errors (omited)
+
+	w.Write(body)
+	w.WriteHeader(http.StatusOK)
+}
+```
+
+To request the resource in the frontend, you can use the methods exposed in the base class `DataSourceWithBackend` (e.g. `getResource` or `postResource`):
+
+```typescript title="src/DataSource.ts"
+export class DataSource extends DataSourceWithBackend<MyQuery, MyDataSourceOptions> {
+  async getTables() {
+    const response = await this.getResource('tables');
+    return response;
+  }
+}
+```
+
+## Summary
+
+This tutorial covered the main steps to convert a frontend data source to a backend data source. The total amount of different use cases is vast so if you have any questions or need help with a specific case, please reach out. Contributions to this guide are also welcome.
```

Comment:
I wanted to add some links here for the "contact" part but not sure where to point to. Thoughts?